README.md

```diff
@@ -1,3 +1,13 @@
 ## Patch
 
 Infrastructure for writing patches which act on other types.
+
+A `Patch` type represents a kind of change made to a datastructure.
+
+```haskell
+class Patch p where
+  type PatchTarget p :: *
+  -- | Apply the patch p a to the value a.  If no change is needed, return
```

Comment:
`@p a@` would look better in Haddock I think, along with `@a@`.

Reply:
Would anything render the readme though? Though that would be neat.

Reply:
Hah didn't realize this was in the README :blush: